graphql-java-spring-webflux/src/main/java/graphql/spring/web/reactive/ExecutionInputCustomizer.java

```diff
@@ -0,0 +1,13 @@
+package graphql.spring.web.reactive;
+
+import graphql.ExecutionInput;
+import graphql.PublicApi;
+import org.springframework.web.server.ServerWebExchange;
+import reactor.core.publisher.Mono;
+
+@PublicApi
+public interface ExecutionInputCustomizer {
+
+    Mono<ExecutionInput> customizeExecutionInput(ExecutionInput executionInput, ServerWebExchange webRequest);
+
+}
```

Comment:
I was thinking it should not be a Mono - but - perhaps it should since you might call out to somewhere to get a special "context token" etc...

So good idea!

Reply:
yea ... it is all about giving the user the possibilities, so I think making it async is a good thing.